services/submission.js

```diff
@@ -1797,38 +1797,66 @@ class Submission {
         return updatedSubmission;
     }
 
-    async _notifyConfigurationChange(userInfo, aSubmission, newModelVersion, prevSubmitter, newSubmitter) {
-        const users = await this.userDAO.getUsersByNotifications([EN.DATA_SUBMISSION.CHANGE_CONFIGURATION]);
+    async _notifyConfigurationChange(aSubmission, newModelVersion, prevSubmitter, newSubmitter) {
         const isSubmitterChanged = Boolean(newSubmitter && prevSubmitter?.id !== newSubmitter?.id);
-        const submitterID = isSubmitterChanged ? newSubmitter?.id : aSubmission?.submitterID;
-        const { submitterEmails, BCCEmails } = (users || []).reduce(
-            (acc, u) => {
-                if (u?.email) {
-                    if (u?.id === submitterID && u.role === USER.ROLES.SUBMITTER) {
-                        acc.submitterEmails.push(u?.email);
-                    }
-
-                    if ([USER.ROLES.FEDERAL_LEAD, USER.ROLES.DATA_COMMONS_PERSONNEL, USER.ROLES.ADMIN].includes(u?.role)) {
-                        acc.BCCEmails.push(u?.email);
-                    }
-                }
-                return acc;
-            },
-            { submitterEmails: [], BCCEmails: [] }
+        // Check if the submitter has the required notification enabled
+        if (!newSubmitter?.notifications?.includes(EN.DATA_SUBMISSION.CHANGE_CONFIGURATION)) {
+            console.warn(`Submission updated; submitter does not have configuration change notifications enabled. submissionID: ${aSubmission?.id}, submitterID: ${newSubmitter?.id}`);
+            return;
+        }
+        // Initialize recipient variables
+        const submitterEmails = [newSubmitter?.email];
+        let CCEmails = [];
+        let BCCEmails = [];
+        // If there's a new submitter, check if the old submitter should be CC'd
+        if (isSubmitterChanged && prevSubmitter?.id) {
+            const prevSubmitterUser = await this.userDAO.findByIdAndStatus(prevSubmitter.id, USER.STATUSES.ACTIVE);
+            if (prevSubmitterUser?.email && 
+                prevSubmitterUser.notifications?.includes(EN.DATA_SUBMISSION.CHANGE_CONFIGURATION) && 
+                prevSubmitterUser.role === USER.ROLES.SUBMITTER) {
```

Comment:
Do we need to check if previous submitter's role? I'd imagine, even when they are not Submitter anymore, we still need to notify them, right?

Reply:
I agree, I don't think we need it. As long as they do have the configuration change notification enabled then I think copying them on the email is ok